libs/cases/backend/case-management/src/lib/services/case.service.ts

```diff
@@ -23,12 +30,15 @@ export class XdCaseService {
 	 */
 	public getAllCases(): Observable<ICaseResponse[]> {
 		return from(this.prismaService.case.findMany()).pipe(
-			map((items) =>
-				items.map((item) => ({
+			map((items) => {
+				if (items.length === 0) {
+					throw new NotFoundException('No Cases found.');
+				}
+				return items.map((item) => ({
 					...item,
 					overdue: Date.now() > new Date(item.dueDate).getTime(),
```

Comment:
For sth like this `defaults` of `lodash` might be better, but it is **not** necessary to change it here, resolve this comment own your own.